extension-as-react-app/src/ExtensionOne.tsx

```diff
@@ -1,33 +1,68 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { useState } from "react";
-// import { useStarships } from "./useStarships";
+import { useStarships } from "./useStarships";
 
 interface ExtensionOneProps {
   onComplete: (object: string) => void;
 }
 
+const minimumNumberOfStarshipSelectionsRequired = 5;
+
 const ExtensionOne: React.FC<ExtensionOneProps> = ({ onComplete }) => {
   const [name, setName] = useState<string>("");
+  const [selectedStarships, setSelectedStarships] = useState<string[]>([]); 
+  const [validationError, setValidationError] = useState<string>("");
   const submitForm = () => {
+    if (selectedStarships.length < minimumNumberOfStarshipSelectionsRequired) {
+      setValidationError('Please select at least 5 starships');
+      return;      
+    }
     console.log(name);
-    onComplete(name);
+    console.log(selectedStarships);
+    setValidationError('');
+    onComplete(
+      selectedStarships.join(','),
+    );
   };
-  // const { starships } = useStarships();
-  // console.log("inside remote", starships);
   const updateName = (inputEvent: any) => {
     setName(inputEvent.target.value);
   };
+  
+
+  const updateSelectedPathways = (inputEvent: any) => {
+    const selectedValues = inputEvent.target.selectedOptions;
+    const selectedValuesArr = Array.prototype.slice.call( selectedValues );
+    console.log(selectedValues);
+    setSelectedStarships(selectedValuesArr.map((option: any) => option.value));
+  }
+ 
+  const { loading, error, starships } = useStarships();
+  if (loading) return <p>Loading...</p>;
+  if (error) return <p>Error!</p>;
+
   return (
     <div>
       <h1 className="text-4xl my-4">Remote: Extension One</h1>
-      <label>Name:</label>
+      <label>Name: </label>
       <input
         type="text"
         placeholder="Please enter your name"
         value={name}
         onChange={updateName}
       />
-      <button type="submit" onClick={submitForm}>
+      <br /><br />
+      <label htmlFor="pathways">Starships:</label>
+      <select name="pathways" id="pathways" onChange={updateSelectedPathways} multiple={true} style={{ "color": "black" }}>
```

Comment:
I added some inline styles but again in real life its how we have decided to styles - tailwind? css modules etc.

Reply:
we've just begun moving from material-ui (v4 😷) to tailwind 🎉